numba/core/config.py

```diff
@@ -337,6 +337,16 @@ def avx_default():
         CUDA_ARRAY_INTERFACE_SYNC = _readenv("NUMBA_CUDA_ARRAY_INTERFACE_SYNC",
                                              int, 1)
 
+        # Path of the directory that the CUDA driver libraries are located
+        CUDA_DRIVER = _readenv("NUMBA_CUDA_DRIVER", str, '')
+
+        # Buffer size for logs produced by CUDA driver operations (e.g.
+        # linking)
+        CUDA_LOG_SIZE = _readenv("NUMBA_CUDA_LOG_SIZE", int, 1024)
+
+        # Whether to generate verbose log messages when JIT linking
+        CUDA_VERBOSE_JIT_LOG = _readenv("NUMBA_CUDA_VERBOSE_JIT_LOG", int, 1)
+
```

Comment:
With moving to `numba.config` I suppose it's possible for users to end up with broken code if they were manipulating the env vars that used to provide this functionality from within python itself. Once `numba.config` is in use, as soon as `numba` is imported I think these configurations are strongly bound to the environment and subsequent manipulation with e.g. `os.environ` will have no effect. The impact is probably minimal and pretty easy to resolve, but just something of which to be aware.

Reply:
Sometimes these variables are read at import time, and possibly sometimes when CUDA is first used - after that I think manipulating them would have no effect anyway... There's a possibility that code manipulating them could get broken by this change, but that code was manipulating undocumented variables that weren't really publicly exposed before now.

Reply:
Yes, I agree, so long as we're aware of it it's fine.